plugins/auth-backend/src/identity/UserInfoDatabaseHandler.test.ts

```diff
@@ -0,0 +1,109 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { resolvePackagePath } from '@backstage/backend-common';
+import { TestDatabaseId, TestDatabases } from '@backstage/backend-test-utils';
+import { Knex } from 'knex';
+import { UserInfoDatabaseHandler } from './UserInfoDatabaseHandler';
+
+const migrationsDir = resolvePackagePath(
+  '@backstage/plugin-auth-backend',
+  'migrations',
+);
+
+describe('UserInfoDatabaseHandler', () => {
+  const databases = TestDatabases.create();
+
+  async function createDatabaseHandler(databaseId: TestDatabaseId) {
+    const knex = await databases.init(databaseId);
+
+    await knex.migrate.latest({
+      directory: migrationsDir,
+    });
+
+    return {
+      knex,
+      dbHandler: new UserInfoDatabaseHandler(knex),
+    };
+  }
+
+  describe.each(databases.eachSupportedId())(
+    '%p',
+    databaseId => {
+      let knex: Knex;
+      let dbHandler: UserInfoDatabaseHandler;
+
+      beforeEach(async () => {
+        ({ knex, dbHandler } = await createDatabaseHandler(databaseId));
+      }, 30000);
+
+      it('addUserInfo', async () => {
+        const userInfo = {
+          claims: {
+            sub: 'user:default/foo',
+            ent: ['group:default/foo-group', 'group:default/bar'],
+            exp: 1234567890,
+          },
+        };
+
+        await dbHandler.addUserInfo(userInfo);
+
+        const savedUserInfo = await knex('user_info')
+          .where('user_entity_ref', 'user:default/foo')
+          .first();
+        expect(savedUserInfo).toEqual({
+          user_entity_ref: 'user:default/foo',
+          user_info: JSON.stringify(userInfo),
+          exp: expect.anything(),
+        });
+
+        userInfo.claims.ent = ['group:default/group1', 'group:default/group2'];
+        await dbHandler.addUserInfo(userInfo);
+
+        const updatedUserInfo = await knex('user_info')
+          .where('user_entity_ref', 'user:default/foo')
+          .first();
+        expect(updatedUserInfo).toEqual({
+          user_entity_ref: 'user:default/foo',
+          user_info: JSON.stringify(userInfo),
+          exp: expect.anything(),
+        });
+      });
+
+      it('getUserInfo', async () => {
+        const userInfo = {
+          claims: {
+            sub: 'user:default/backstage-user',
+            ent: ['group:default/group1', 'group:default/group2'],
+            exp: 1234567890,
+          },
+        };
+
+        await knex('user_info').insert({
+          user_entity_ref: 'user:default/backstage-user',
+          user_info: JSON.stringify(userInfo),
+          exp: knex.fn.now(),
+        });
+
+        const savedUserInfo = await dbHandler.getUserInfo(
+          'user:default/backstage-user',
+        );
+        expect(savedUserInfo).toEqual(userInfo);
+      });
+    },
+    60000,
```

Comment:
if you want, you can do `jest.setTimeout(60_000);` at the top of the file instead - it's much easier to not forget then, when adding more tests in the same file :) And then you don't need the timeout on beforeEach either, I think.

Reply:
ah thanks for the tip!